src/rtk/features/apiDelegation/delegableApi/delegableApiSlice.ts

```diff
@@ -209,10 +232,61 @@ const delegableApiSlice = createSlice({
         } else {
           state.error.message = 'Unknown error';
         }
+      })
+      .addCase(apiDelegationCheck.pending, (state, action) => {
+        const apiIdentifier = action.meta.arg.right.value;
+
+        state.presentedApiList = state.presentedApiList.map((api: DelegableApi) => {
+          if (api.identifier === apiIdentifier) {
+            api.isLoading = true;
+          }
+          return api;
+        });
+      })
+      .addCase(apiDelegationCheck.fulfilled, (state, action) => {
+        const dto: DelegationCheckDto = action.meta.arg;
+        const apiIdentifier = dto.right.value;
+
+        if (action.payload[0].status === 'Delegable') {
+          state.delegableApiList = state.delegableApiList.filter(
+            (delegableApi) => dto.delegableApi.identifier !== delegableApi.identifier,
+          );
+          state.presentedApiList = state.presentedApiList.filter(
+            (delegableApi) => dto.delegableApi.identifier !== delegableApi.identifier,
+          );
+          state.delegableApiSearchPool = state.delegableApiSearchPool.filter(
+            (delegableApi) => dto.delegableApi.identifier !== delegableApi.identifier,
+          );
+
+          state.chosenDelegableApiList.push(dto.delegableApi);
+        } else {
+          const nextStateArray: DelegableApi[] = state.presentedApiList.map((api: DelegableApi) => {
+            if (api.identifier === apiIdentifier) {
+              api.isLoading = false;
+              api.errorCode = action.payload[0].details[0].code;
+            }
+            return api;
+          });
+
+          state.presentedApiList = nextStateArray;
```

Comment:
Note: Since changes are only made to the presentedApiList, and not to the list the presented list is created from, every time an api is hidden due to changes in search or filters, it will 'forget' that it was red and undelegable.

That isn't necessarily unacceptable behaviour, but it is also not ideal.
A more stable solution might be to make changes to the underlying delegableApiList and then trigger a new search.